### PR TITLE
list_insert now works as it should

### PR DESCRIPTION
**Pull request by taiyu-len**
also put list resizing into a static function.

### CODE REVIEW

File: sway/list.c [hunk 1]
```diff
@@ -11,6 +11,14 @@ list_t *create_list(void) {
 	return list;
 }
 
+static
+void list_resize(list_t *list) {
```
Comment (SirCmpwn) [inline sway/list.c]: Style issue - join these two lines

File: sway/list.c [hunk 2]
```diff
@@ -20,25 +28,20 @@ void list_free(list_t *list) {
...
+	list_resize(list);
+	memmove(&list->items[index + 1], &list->items[index], sizeof(void*) * (list->length - index));
+	list->length++;
+	list->items[index] = item;
 }
 
 void list_del(list_t *list, int index) {
 	list->length--;
-	memmove(&list->items[index], &list->items[index + 1], sizeof(void*) * (list->capacity - index - 1));
+	memmove(&list->items[index], &list->items[index + 1], sizeof(void*) * (list->length - index));
```
Comment (SirCmpwn) [inline sway/list.c]: Using `length` here is a waste of cycles. If we memmove with the capacity, we move fewer items (leaving garbage from `items[list->length ... list->capacity]`).

Comment (taiyu-len) [inline sway/list.c]: capacity is the one that moves more though.
length is always less then capacity, so it moves less.

Comment (SirCmpwn) [inline sway/list.c]: Oh, my bad, I read the diff backwards.

Comment (taiyu-len) [inline sway/list.c]: no problem, i wanted to try to implement the multikey handling with list_t but compiler complains about int->void* void*->int to much.
ill send another pr with it edited a bit later.

**Pull request merged**